### ISSUE TITLE
[google_maps_flutter] java.lang.SecurityException: GoogleCertificatesRslt: not allowed exception: pkg={package_name}

### ISSUE THREAD

**Issue by cybex-dev** [flutter/flutter#109115]
<details>
  <summary>Flutter Doctor</summary>
 
```
[✓] Flutter (Channel stable, 3.0.1, on Arch Linux 5.18.12-arch1-1, locale en_US.UTF-8)
    • Flutter version 3.0.1 at /home/cybex/.flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (3 months ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc1)
    • Android SDK at /home/cybex/Android/Sdk
    • Platform android-33, build-tools 33.0.0-rc1
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /usr/bin/google-chrome-stable

[✓] Linux toolchain - develop for Linux desktop
    • clang version 14.0.6
    • cmake version 3.23.2
    • ninja version 1.11.0
    • pkg-config version 1.8.0

[✓] Android Studio (version 2021.2)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 69.0.2
    • Dart plugin version 212.5744
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3)
    • IntelliJ at /home/cybex/.local/share/JetBrains/Toolbox/apps/IDEA-U/ch-0/213.6777.52
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (3 available)
    • SM N975U1 (mobile) • RF8MC02F23L • android-arm64  • Android 12 (API 31)
    • Linux (desktop)    • linux       • linux-x64      • Arch Linux 5.18.12-arch1-1
    • Chrome (web)       • chrome      • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Problem:**

There does not appear to be much information about this `java.lang.SecurityException: GoogleCertificatesRslt: not allowed:` exception nor resolutions for it and it appears to be unique to this package.

_Following steps to setup/configure Google Maps integration into Flutter results in an exception. The maps object still show tiles and is usable, however this exception should be addressed._ 

**Project Setup:**

 1. Add to pubspec (https://pub.dev/packages/google_maps_flutter#getting-started)

**pubspec.yaml**

```yaml
environment:
  sdk: ">=2.17.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  ...

  google_maps_flutter: ^2.1.10
```

 2. Enable & Generate API keys from [Google Maps SDK](https://pub.dev/packages/google_maps_flutter#getting-started) (restrict with SHA256 from signingTool & playstore [android] or bundleId [iOS)

_**Note**: SHA256 & SHA1 hashes are used for Firebase login (with phone) which work as intended._

 4. Set [minSdkVersion](https://pub.dev/packages/google_maps_flutter#android) 
 
**android/app/build.gradle**
```
android {
    defaultConfig {
            applicationId "my.awesome.app"
            minSdkVersion 20
            targetSdkVersion flutter.targetSdkVersion
            ...
    }
}
```
 5. Add [API_KEY](https://pub.dev/packages/google_maps_flutter#android) (android) 

**AndroidManifest.xml**
```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="my.awesome.app">

    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />

    <meta-data
        android:name="com.google.android.geo.API_KEY"
        android:value="{restricted Android Maps API key}" />

    <application
        android:name="${applicationName}"
        android:icon="@mipmap/ic_launcher"
        android:label="My Awesome App">

        <activity.../>

       ...
  
    </application>
</manifest>
```

**Implementation (Android)**

```dart
class _Maps extends StatelessWidget {
  const _Maps({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
   final googleMapController = Completer<GoogleMapController>();

    return GoogleMap(
      mapType: MapType.hybrid,
      initialCameraPosition: _kGooglePlex,
      onMapCreated: (GoogleMapController controller) => googleMapController.complete(controller),
    );
  }
}
```

Results in the following error/exception.

**Exception:**
```
W/DynamiteModule( 9849): Local module descriptor class for com.google.android.gms.googlecertificates not found.
I/DynamiteModule( 9849): Considering local module com.google.android.gms.googlecertificates:0 and remote module com.google.android.gms.googlecertificates:7
I/DynamiteModule( 9849): Selected remote version of com.google.android.gms.googlecertificates, version >= 7
I/SnapshotHandler( 9849): Unable to retrieve flag snapshot for com.google.android.libraries.consentverifier#my.awesome.app, using defaults.
W/MobStoreFlagStore( 9849): Unable to update local snapshot for com.google.android.libraries.consentverifier#my.awesome.app, may result in stale flags.
W/MobStoreFlagStore( 9849): java.util.concurrent.ExecutionException: java.lang.SecurityException: GoogleCertificatesRslt: not allowed: pkg=my.awesome.app, sha256=[valid-sha256-hash], atk=false, ver=222615044.true (go/gsrlt)
W/MobStoreFlagStore( 9849): 	at ahv.s(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):3)
W/MobStoreFlagStore( 9849): 	at ahv.get(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):2)
W/MobStoreFlagStore( 9849): 	at ajf.g(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):2)
W/MobStoreFlagStore( 9849): 	at xx.d(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):1)
W/MobStoreFlagStore( 9849): 	at xz.run(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):0)
W/MobStoreFlagStore( 9849): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
W/MobStoreFlagStore( 9849): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/MobStoreFlagStore( 9849): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/MobStoreFlagStore( 9849): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/MobStoreFlagStore( 9849): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/MobStoreFlagStore( 9849): 	at java.lang.Thread.run(Thread.java:920)
W/MobStoreFlagStore( 9849): Caused by: java.lang.SecurityException: GoogleCertificatesRslt: not allowed: pkg=my.awesome.app, sha256=[valid-sha256-hash], atk=false, ver=222615044.true (go/gsrlt)
W/MobStoreFlagStore( 9849): 	at android.os.Parcel.createExceptionOrNull(Parcel.java:2437)
W/MobStoreFlagStore( 9849): 	at android.os.Parcel.createException(Parcel.java:2421)
W/MobStoreFlagStore( 9849): 	at android.os.Parcel.readException(Parcel.java:2404)
W/MobStoreFlagStore( 9849): 	at android.os.Parcel.readException(Parcel.java:2346)
W/MobStoreFlagStore( 9849): 	at em.bc(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):2)
W/MobStoreFlagStore( 9849): 	at rj.a(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):4)
W/MobStoreFlagStore( 9849): 	at it.e(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):2)
W/MobStoreFlagStore( 9849): 	at js.t(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):2)
W/MobStoreFlagStore( 9849): 	at js.u(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):3)
W/MobStoreFlagStore( 9849): 	at js.e(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):2)
W/MobStoreFlagStore( 9849): 	at jw.handleMessage(:com.google.android.gms.dynamite_mapsdynamite@222615097@22.26.15 (190400-0):69)
W/MobStoreFlagStore( 9849): 	at android.os.Handler.dispatchMessage(Handler.java:102)
W/MobStoreFlagStore( 9849): 	at android.os.Looper.loopOnce(Looper.java:226)
W/MobStoreFlagStore( 9849): 	at android.os.Looper.loop(Looper.java:313)
W/MobStoreFlagStore( 9849): 	at android.os.HandlerThread.run(HandlerThread.java:67)
```

**Comment by kapilb145** [comment]
I am also facing this problem. anyone please share solution for the same.

**Comment by maheshmnj** [comment]
Hi @cybex-dev, Thanks for filing the issue. I am able to reproduce the issue on the latest stable and the master channel. Upon quick search I landed on this issue tracker https://issuetracker.google.com/issues/228091313?pli=1, So I think this issue doesn't seem to be from the plugin, But from the Google services, But I may be wrong, Labeling this issue for further insights from the team.

<details>
<summary>logs</summary>

```
Could not load custom device from config index 0: Expected enabled to be a boolean.
Running "flutter pub get" in sample...
Launching lib/main.dart on Redmi K20 Pro in debug mode...
Note: /Users/mahesh/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-2.1.10/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:65195/ljMiJj27KfY=/ws
W/BpBinder(25028): PerfMonitor binderTransact :  time=326ms interface=android.view.IWindowSession code=5
W/Looper  (25028): PerfMonitor doFrame : time=333ms vsyncFrame=2338631 latency=1ms procState=2
D/MapsInitializer(25028): preferredRenderer: null
D/zzca    (25028): preferredRenderer: null
I/zzca    (25028): Making Creator dynamically
I/DynamiteModule(25028): Considering local module com.google.android.gms.maps_dynamite:0 and remote module com.google.android.gms.maps_dynamite:203115000
I/DynamiteModule(25028): Selected remote version of com.google.android.gms.maps_dynamite, version >= 203115000
V/DynamiteModule(25028): Dynamite loader version >= 2, using loadModule2NoCrashUtils
W/System  (25028): ClassLoader referenced unknown path:
W/.example.sampl(25028): ClassLoaderContext classpath size mismatch. expected=4, found=1 (DLC[];PCL[/data/app/~~L5DUgCsiqN801GzjXtNE5g==/com.whatsapp-7750a_c4Qkt-Yu9TCV-gCA==/base.apk*1325529428:/data/app/~~L5DUgCsiqN801GzjXtNE5g==/com.whatsapp-7750a_c4Qkt-Yu9TCV-gCA==/base.apk!classes2.dex*2747456886:/data/app/~~L5DUgCsiqN801GzjXtNE5g==/com.whatsapp-7750a_c4Qkt-Yu9TCV-gCA==/base.apk!classes3.dex*3233097562:/data/app/~~L5DUgCsiqN801GzjXtNE5g==/com.whatsapp-7750a_c4Qkt-Yu9TCV-gCA==/base.apk!classes4.dex*2860009861]{PCL[/system/framework/org.apache.http.legacy.jar*1721227950]} | DLC[];PCL[/data/app/~~fX7E1TsanF4r86EXL8gz8A==/com.example.sample-p0cvUytg_b-nLwqm32BK7Q==/base.apk*981096004]{PCL[/system/framework/org.apache.http.legacy.jar*1721227950]})
W/.example.sampl(25028): Found duplicate classes, falling back to extracting from APK : /data/user_de/0/com.google.android.gms/app_chimera/m/0000032c/MapsDynamite.apk
W/.example.sampl(25028): NOTE: This wastes RAM and hurts startup performance.
W/.example.sampl(25028): Found duplicated class when checking oat files: 'Landroid/support/v4/app/RemoteActionCompatParcelizer;' in /data/app/~~fX7E1TsanF4r86EXL8gz8A==/com.example.sample-p0cvUytg_b-nLwqm32BK7Q==/base.apk and /data/user_de/0/com.google.android.gms/app_chimera/m/0000032c/MapsDynamite.apk
I/Google Maps Android API(25028): Google Play services client version: 12451000
I/SnapshotHandler(25028): Unable to retrieve flag snapshot for com.google.android.gms.maps#com.example.sample, using defaults.
I/Google Maps Android API(25028): Google Play services package version: 222615037
I/Google Maps Android API(25028): Google Play services maps renderer version(legacy): 203115000
D/MapsInitializer(25028): loadedRenderer: LEGACY
D/zzca    (25028): preferredRenderer: null
I/Google Maps Android API(25028): Google Play services package version: 222615037
I/Google Maps Android API(25028): Google Play services maps renderer version(legacy): 203115000
I/TetheringManager(25028): registerTetheringEventCallback:com.example.sample
W/Google Maps Android API(25028): Deprecation notice: In a future release, indoor will no longer be supported on satellite, hybrid or terrain type maps. Even where indoor is not supported, isIndoorEnabled() will continue to return the value that has been set via setIndoorEnabled(), as it does now. By default, setIndoorEnabled is 'true'. The API release notes (https://developers.google.com/maps/documentation/android-api/releases) will let you know when indoor support becomes unavailable on those map types.
I/PlatformViewsController(25028): Hosting view in a virtual display for platform view: 0
I/Choreographer(25028): Skipped 30 frames!  The application may be doing too much work on its main thread.
I/.example.sampl(25028): ProcessProfilingInfo new_methods=1380 is saved saved_to_disk=1 resolve_classes_delay=8000
D/SurfaceView(25028): UPDATE null, mIsCastMode = false
W/Looper  (25028): PerfMonitor doFrame : time=80ms vsyncFrame=2338741 latency=502ms procState=2 historyMsgCount=1 (msgIndex=1 wall=516ms seq=113 running=323ms runnable=29ms io=41ms binder=121ms h=android.os.Handler c=io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0)
E/GoogleMapController(25028): Cannot enable MyLocation layer as location permissions are not granted
I/ba      (25028): Successfully registered with Phenotype.
W/BpBinder(25028): PerfMonitor binderTransact :  time=562ms interface=com.google.android.gms.maps.auth.IApiTokenService code=1
E/libc    (25028): Access denied finding property "ro.vendor.df.effect.conflict"
E/libc    (25028): Access denied finding property "ro.vendor.knock.type"
W/Thread-6(25028): type=1400 audit(0.0:289830): avc: denied { read } for name="u:object_r:vendor_displayfeature_prop:s0" dev="tmpfs" ino=29893 scontext=u:r:untrusted_app:s0:c112,c260,c512,c768 tcontext=u:object_r:vendor_displayfeature_prop:s0 tclass=file permissive=0 app=com.example.sample
W/DynamiteModule(25028): Local module descriptor class for com.google.android.gms.googlecertificates not found.
I/DynamiteModule(25028): Considering local module com.google.android.gms.googlecertificates:0 and remote module com.google.android.gms.googlecertificates:7
I/DynamiteModule(25028): Selected remote version of com.google.android.gms.googlecertificates, version >= 7
I/SnapshotHandler(25028): Unable to retrieve flag snapshot for com.google.android.libraries.consentverifier#com.example.sample, using defaults.
W/MobStoreFlagStore(25028): Unable to update local snapshot for com.google.android.libraries.consentverifier#com.example.sample, may result in stale flags.
W/MobStoreFlagStore(25028): java.util.concurrent.ExecutionException: java.lang.SecurityException: GoogleCertificatesRslt: not allowed: pkg=com.example.sample, sha256=[7977edf4787b8f62b4ba85af96f3c847404ee6ed8f93f9b1b92080dc7a8fcba7], atk=false, ver=222615037.true (go/gsrlt)
W/MobStoreFlagStore(25028): 	at ahv.s(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):3)
W/MobStoreFlagStore(25028): 	at ahv.get(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):2)
W/MobStoreFlagStore(25028): 	at ajf.g(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):2)
W/MobStoreFlagStore(25028): 	at xx.d(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):1)
W/MobStoreFlagStore(25028): 	at xz.run(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):0)
W/MobStoreFlagStore(25028): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
W/MobStoreFlagStore(25028): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/MobStoreFlagStore(25028): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/MobStoreFlagStore(25028): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/MobStoreFlagStore(25028): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/MobStoreFlagStore(25028): 	at java.lang.Thread.run(Thread.java:923)
W/MobStoreFlagStore(25028): Caused by: java.lang.SecurityException: GoogleCertificatesRslt: not allowed: pkg=com.example.sample, sha256=[7977edf4787b8f62b4ba85af96f3c847404ee6ed8f93f9b1b92080dc7a8fcba7], atk=false, ver=222615037.true (go/gsrlt)
W/MobStoreFlagStore(25028): 	at android.os.Parcel.createExceptionOrNull(Parcel.java:2376)
W/MobStoreFlagStore(25028): 	at android.os.Parcel.createException(Parcel.java:2360)
W/MobStoreFlagStore(25028): 	at android.os.Parcel.readException(Parcel.java:2343)
W/MobStoreFlagStore(25028): 	at android.os.Parcel.readException(Parcel.java:2285)
W/MobStoreFlagStore(25028): 	at em.bc(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):2)
W/MobStoreFlagStore(25028): 	at rj.a(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):4)
W/MobStoreFlagStore(25028): 	at it.e(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):2)
W/MobStoreFlagStore(25028): 	at js.t(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):2)
W/MobStoreFlagStore(25028): 	at js.u(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):3)
W/MobStoreFlagStore(25028): 	at js.e(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):2)
W/MobStoreFlagStore(25028): 	at jw.handleMessage(:com.google.android.gms.dynamite_mapsdynamite@222615081@22.26.15 (150400-0):69)
W/MobStoreFlagStore(25028): 	at android.os.Handler.dispatchMessage(Handler.java:102)
W/MobStoreFlagStore(25028): 	at android.os.Looper.loop(Looper.java:236)
W/MobStoreFlagStore(25028): 	at android.os.HandlerThread.run(HandlerThread.java:67)
I/Counters(25028): exceeded sample count in FrameTime
I/Counters(25028): exceeded sample count in FrameTime
I/chatty  (25028): uid=11136(com.example.sample) androidmapsapi- identical 3 lines
I/Counters(25028): exceeded sample count in FrameTime
I/Counters(25028): exceeded sample count in FrameTime
I/chatty  (25028): uid=11136(com.example.sample) androidmapsapi- identical 6 lines
I/Counters(25028): exceeded sample count in FrameTime
I/Counters(25028): exceeded sample count in FrameTime
I/chatty  (25028): uid=11136(com.example.sample) androidmapsapi- identical 8 lines
I/Counters(25028): exceeded sample count in FrameTime
I/Counters(25028): exceeded sample count in FrameTime
I/chatty  (25028): uid=11136(com.example.sample) androidmapsapi- identical 8 lines
I/Counters(25028): exceeded sample count in FrameTime
I/Counters(25028): exceeded sample count in FrameTime
I/Counters(25028): exceeded sample count in FrameTime
I/chatty  (25028): uid=11136(com.example.sample) androidmapsapi- identical 7 lines
I/Counters(25028): exceeded sample count in FrameTime
I/Counters(25028): exceeded sample count in FrameTime
I/chatty  (25028): uid=11136(com.example.sample) androidmapsapi- identical 6 lines
I/Counters(25028): exceeded sample count in FrameTime

```
</details>   
<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.5 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 days ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.2009, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.1.0-0.0.pre.2009 on channel master at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9d96c7b8be (3 days ago), 2022-08-01 00:39:06 -0400
    • Engine revision cf0db3e546
    • Dart version 2.19.0 (build 2.19.0-49.0.dev)
    • DevTools version 2.16.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.69.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 104.0.5112.79

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by gilmarwsr** [comment]
Hi @cybex-dev , I really don't know if this is related but let me tell you how I reached this issue, and maybe It can help us to understand what is happening.

I'm trying to intercept traffic using an TLS/SSL proxy. I'm using Frida to bypass security controls like certificate pinning (Its for cybersecurity research purposes). I've been able to "resolve" most of pinning issues, but the application is using Recaptcha (google service) to combat possible brute-force on a specific endpoint... and I'm thinking this is the only pinning/certificate transparency or something related issue that I can't handle.
I didn't receive any call on my TLS proxy, but the application suddenly closes the Recaptcha and show an generic communication error to the user.
In logcat I received this message:

08-11 13:10:07.543 13327 13407 E bgqz    : GoogleCertificatesRslt: not allowed: pkg=[my package], sha256=[], atk=false, ver=222615030.true (go/gsrlt): null [CONTEXT service_id=247 ]

**Comment by SolitaryWolf** [comment]
Same for me. My map cannot be dragged. any update?

**Comment by JanVilimek** [comment]
Same error, still not working.  - GoogleCertificatesRslt: not allowed: pkg=...

**Comment by Rashid-Khabeer** [comment]
Same error, I am using google maps in my app and I have to publish it, but due to this issue I am unable to publish my app, the app crashes when I open the google maps page.

java.lang.SecurityException: GoogleCertificatesRslt:

@maheshmnj

**Comment by Tom3652** [comment]
I have the same error despite i have setup correctly my API key restrictions

**Comment by trungdn0401** [comment]
Same error, any news on this please?

**Comment by xmany** [comment]
Same error

**Comment by xmany** [comment]
The map feels significantly more jittery when dragging

**Comment by codendtech** [comment]
sam error, but map is working

**Comment by Bikuta-victor1** [comment]
same error 
any fix yet

**Comment by VirusXMe** [comment]
same here

**Comment by ClaudeChey** [comment]
same error

**Comment by 3nd3rs0nn** [comment]
Same error!

**Comment by eexpresstam** [comment]
Same error here.

**Comment by nasir447** [comment]
Same issue here, any solution yet?

**Comment by nasir447** [comment]
> sam error, but map is working

map working in old devices or older versions of android. I have a physical device of android 9, and map is working in it but in a second device of Android 12, it is not working

**Comment by smhoz** [comment]
It's work for me 
implementation 'com.google.android.gms:play-services-location:19.0.1'

**Comment by nasir447** [comment]
@cybex-dev did you find any workaround for this?

**Comment by lapisanlangit** [comment]
same error..bu i am using ionic capacitor

**Comment by wamynobe** [comment]
same error here

**Comment by johnliedtke** [comment]
Are there any workarounds here someone can share?

**Comment by ArlysthonFeitosa** [comment]
java.util.concurrent.ExecutionException: java.lang.SecurityException: GoogleCertificatesRslt: not allowed: pkg=com.g4flex.flex4company, sha256=[MY-SHA256-HERE], atk=false, ver=224113044.true (go/gsrlt)


that was the error i've got.
so i added the provided SHA256 key to the firebase SDK config.
it worked

**Comment by joseph-lewis** [comment]
Where is the Firebase SDK config @ArlysthonFeitosa ? Can you be more specific, as its an ongoing issue that hasn't been resolved.

**Comment by ArlysthonFeitosa** [comment]
i thought that it was solved, but no :(

**Comment by rahmanrezaee** [comment]
any update???

**Comment by parthh2** [comment]
Any update?

**Comment by murodjon5117** [comment]
Same problem

**Comment by Hesham-Dev-LY** [comment]
Same **problem** here

**Comment by norbdev** [comment]
Same problem here, but with native Android, latest Play Services, maps and location lib. Map working fine but logcat prints the error

**Comment by sumanwebidigital** [comment]
Same problem

**Comment by Gopiraju-swensa** [comment]
same problem

**Comment by emeban44** [comment]
same problem

**Comment by rraayy** [comment]
same problem needs help!!

**Comment by mhmzdev** [comment]
Any update on this?

**Comment by xdelvalle** [comment]
Same error but with ionic capacitor, Any updates?

**Comment by moazelsawaf** [comment]
Facing the same issue.

**Comment by lanrehnics** [comment]
up

**Comment by hgandhi67** [comment]
same here

**Comment by hgandhi67** [comment]
I got the same issue in android only and after researching on google i found out that after enabling 
**Maps SDK for Android** API on https://console.cloud.google.com/google/maps-apis/ issue was resolved and maps started loading without any code change.

**Comment by zahidshaikh08** [comment]
I got the same issue in ANDROID ONLY after researching on google i found out that after enabling 
**'Maps SDK for Android API'** on https://console.cloud.google.com/google/maps-apis/ issue was resolved and maps started loading without any code change.

**Comment by maheshmnj** [comment]
> same problem
> Facing the same issue ...

Please avoid comments like these, They do not provide any value to the issue, But instead, just ping everyone unnecessarily in the thread.

**Comment by exaby73** [comment]
I can still reproduce this issue on `Master (3.7.0-3.0.pre.37)`. I have restricted the API with my SHA-1 debug key.

<details>
<summary>Code Sample</summary>

```dart
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Center(
        child: _Maps(),
      ),
    );
  }
}

class _Maps extends StatelessWidget {
  const _Maps({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final googleMapController = Completer<GoogleMapController>();

    return GoogleMap(
      mapType: MapType.hybrid,
      initialCameraPosition: const CameraPosition(target: LatLng(0, 0)),
      onMapCreated: (GoogleMapController controller) =>
          googleMapController.complete(controller),
    );
  }
}
```

</details>

<details>
<summary>Error logs</summary>

```console
W/MobStoreFlagStore( 5422): Caused by: java.lang.SecurityException: GoogleCertificatesRslt: not allowed: pkg=com.example.issue_109115, sha256=[43fe49f69c6fc547842cf70697e5e03513610ff9c4ff3590a4dfc15c47e6f44c], atk=true, ver=221821047.true (go/gsrlt)
W/MobStoreFlagStore( 5422): 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3011)
W/MobStoreFlagStore( 5422): 	at android.os.Parcel.createException(Parcel.java:2995)
W/MobStoreFlagStore( 5422): 	at android.os.Parcel.readException(Parcel.java:2978)
W/MobStoreFlagStore( 5422): 	at android.os.Parcel.readException(Parcel.java:2920)
W/MobStoreFlagStore( 5422): 	at ei.ba(:com.google.android.gms.dynamite_mapsdynamite@221821104@22.18.21 (190800-0):2)
```

</details>

**Comment by paulcappadona** [comment]
This appears to be solved now https://github.com/flutter/plugins/pull/6619
Available in google_maps_flutter 2.2.2

**Comment by azaderdogan** [comment]
i am also having the same problem. what should I do?

**Comment by SmokeDevz** [comment]
check permissions app, this work's for me

**Comment by IdrissMiraoui** [comment]
I have the same error here !
Any update please

**Comment by iSWATxJOKERi** [comment]
Same here 😞

**Comment by exaby73** [comment]
Hello guys. Instead of commenting "Same", please upvote the [original comment](https://github.com/flutter/flutter/issues/109115#issue-1331029959) with a thumbs up emoji. The team [uses this as a metric to prioritize issues](https://github.com/flutter/flutter/issues/109115#issue-1331029959). Thank you

**Comment by malkomich** [comment]
It happened to me trying in the Android Emulator.

As soon as I tried in a real device, the error was not showing up.

It looks to be, at least on my case, to the missing Google Play SDK, and probably a Google account configured in the device, to access APIs like Google Maps.

**Comment by adiiisanjayaa** [comment]
any update?

**Comment by stuartmorgan** [comment]
Closing as out of scope; per https://issuetracker.google.com/issues/228091313#comment19 this is an issue in the Google Maps SDK itself, not with the Flutter plugin, and thus isn't something the Flutter team can fix.

The comments there indicate that the new renderer does not have this logging issue; see https://pub.dev/packages/google_maps_flutter_android#map-renderer for instructions on enabling it in Flutter.

**Comment by alimcevik** [comment]
Actually I realized that the problem is related to API keys. I was getting the same error in a Flutter project using Firebase.

After running the application with the key on the Google Cloud Console for the solution, I saw a change when I downloaded the google-services.json file in Firebase again.
The problem is now fixed for me.
